interfaces.go

```diff
@@ -93,10 +93,13 @@ func (b *Breadcrumb) MarshalJSON() ([]byte, error) {
 // User describes the user associated with an Event. If this is used, at least
 // an ID or an IP address should be provided.
 type User struct {
-	Email     string `json:"email,omitempty"`
-	ID        string `json:"id,omitempty"`
-	IPAddress string `json:"ip_address,omitempty"`
-	Username  string `json:"username,omitempty"`
+	Data      map[string]string `json:"data,omitempty"`
+	Email     string            `json:"email,omitempty"`
+	ID        string            `json:"id,omitempty"`
+	IPAddress string            `json:"ip_address,omitempty"`
+	Name      string            `json:"name,omitempty"`
+	Segment   string            `json:"segment,omitempty"`
```

Comment:
Could we table test this to make sure that `omitempty` is correctly set and being used?

https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

Reply:
Add a new test `TestUserMarhsalJson`. Let me know if this what you had in mind